app/test_main.py

```diff
@@ -1 +1,63 @@
-# write your code here
+import pytest
+
+from _pytest.monkeypatch import MonkeyPatch
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, expected_status",
+    [
+        ("https://www.google.com", "Not accessible"),
+        ("https://www.google.com", "Not accessible"),
+        ("https://www.google.com", "Not accessible")
+    ]
+)
+def test_cannot_access_if_connection_or_valid_url_is_true(
+        monkeypatch: MonkeyPatch,
+        url: str,
+        expected_status: str
+) -> None:
+    monkeypatch.setattr(
+        "app.main.has_internet_connection", lambda: False
+    )
+    monkeypatch.setattr(
+        "app.main.valid_google_url", lambda url: False
+    )
+    assert can_access_google_page(url) == expected_status
+
+
+@pytest.mark.parametrize(
+    "url, expected_status",
+    [
+        ("https://www.google.com", "Not accessible"),
+        ("https://www.google.com", "Not accessible")
+    ]
+)
+def test_cannot_access_if_only_connection(
+        monkeypatch: MonkeyPatch,
+        url: str,
+        expected_status: str
+) -> None:
+    monkeypatch.setattr(
+        "app.main.has_internet_connection", lambda: False
+    )
+    assert can_access_google_page(url) == expected_status
+
+
+@pytest.mark.parametrize(
+    "url, expected_status",
+    [
+        ("https://www.google.com", "Not accessible"),
+        ("https://www.google.com", "Not accessible")
+    ]
+)
+def test_cannot_access_if_only_valid_url(
+        monkeypatch: MonkeyPatch,
+        url: str,
+        expected_status: str
+) -> None:
+    monkeypatch.setattr(
+        "app.main.valid_google_url", lambda url: False
+    )
+    assert can_access_google_page(url) == expected_status
```

Comment:
Pass mocks as decorators to a function using `@mock.patch`. Create one function and use two boolean values for mocks and an expected result as a parameter that is passed. Your first parameter to the ` @pytest.mark.parametrize` should be like `"valid_url_value, connection_value, expected_value"`.